ts/indicators/pivot-points.src.ts

```diff
@@ -0,0 +1,483 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import H from '../parts/Globals.js';
+
+/**
+ * Internal types
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        class PivotPointsIndicator extends SMAIndicator {
+            public camarillaPlacement(
+                values: Array<number>
+            ): Array<number>;
+            public data: Array<PivotPointsIndicatorPoint>;
+            public endPoint: number;
+            public fibonacciPlacement(
+                values: Array<number>
+            ): Array<(number|null)>;
+            public getPivotAndHLC(
+                values: Array<Array<number>>
+            ): [number, number, number, number];
+            public getValues(
+                series: Series,
+                params: PivotPointsIndicatorParamsOptions
+            ): (boolean|IndicatorMultipleValuesObject);
+            public nameBase: string;
+            public options: PivotPointsIndicatorOptions;
+            public plotEndPoint: number;
+            public pointArrayMap: Array<string>;
+            public pointClass: typeof PivotPointsIndicatorPoint;
+            public points: Array<PivotPointsIndicatorPoint>;
+            public pointValKey: string;
+            public standardPlacement(
+                values: Array<number>
+            ): Array<(number|null)>
+            public toYData(point: Point): Array<number>;
+            public translate(): void;
+        }
+
+        interface PivotPointsIndicatorParamsOptions
+            extends SMAIndicatorParamsOptions {
+            algorithm?: string;
+        }
+
+        class PivotPointsIndicatorPoint extends SMAIndicatorPoint {
+            public P: number;
+            public pivotLine: string;
+            public series: PivotPointsIndicator;

```

Comment:
The `PivotPointsIndicatorPoint` class is lacking the `destroy` and `destroyElements` functions, that are defined below.

Reply:
But aren't these defined by Point and thus declared and available through inheritance?

Reply:
No, only the type is defined by inheritence, but the implementation is not. You overwrite the inherited functions. Therefor you have to define it again, so that inheritence conflicts in future changes show up on the right spot.

Reply:
Added missing function declarations to `PivotPointsIndicatorPoint`